src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+    @Nullable
+    public IEntityLivingData onInitialSpawn(final DifficultyInstance difficulty, @Nullable final IEntityLivingData living)
+    {
+        final IEntityLivingData livingdata = super.onInitialSpawn(difficulty, living);
+        this.setEquipmentBasedOnDifficulty(difficulty);
+        this.setEnchantmentBasedOnDifficulty(difficulty);
+        this.setCombatTask();
+        this.setCanPickUpLoot(this.rand.nextFloat() < 0.55F * difficulty.getClampedAdditionalDifficulty());
+
+        if (this.getItemStackFromSlot(EntityEquipmentSlot.HEAD).isEmpty())
+        {
+            final Calendar calendar = this.world.getCurrentDate();
+
+            if (calendar.get(2) + 1 == 10 && calendar.get(5) == 31 && this.rand.nextFloat() < 0.25F)
+            {
+                this.setItemStackToSlot(EntityEquipmentSlot.HEAD, new ItemStack(this.rand.nextFloat() < 0.1F ? Blocks.LIT_PUMPKIN : Blocks.PUMPKIN));
+                this.inventoryArmorDropChances[EntityEquipmentSlot.HEAD.getIndex()] = 0.0F;
+            }
+        }
+
+        return livingdata;
+    }
+
+    /**
+     * sets this entity's combat AI.
+     */
+    public void setCombatTask()
+    {
+        if (this.world != null && !this.world.isRemote)
+        {
+            this.tasks.removeTask(this.aiAttackOnCollide);
+            this.tasks.removeTask(this.aiArrowAttack);
+            final ItemStack itemstack = this.getHeldItemMainhand();
+
+            if (itemstack.getItem() == Items.BOW)
+            {
+                int i = BOW_DAMAGE;
+
+                if (this.world.getDifficulty() != EnumDifficulty.HARD)
+                {
+                    i = BOW_DAMAGE_HARD;
+                }
+
+                this.aiArrowAttack.setAttackCooldown(i);
+                this.tasks.addTask(4, this.aiArrowAttack);
+            }
+            else
+            {
+                this.tasks.addTask(4, this.aiAttackOnCollide);
+            }
+        }
+    }
+
+    /**
+     * Attack the specified entity using a ranged attack.
+     */
+    public void attackEntityWithRangedAttack(final EntityLivingBase target, final float distanceFactor)
+    {
+        final EntityArrow entityarrow = this.getArrow(distanceFactor);
+        final double d0 = target.posX - this.posX;
+        final double d1 = target.getEntityBoundingBox().minY + (double) (target.height / 3.0F) - entityarrow.posY;
+        final double d2 = target.posZ - this.posZ;
+        final double d3 = (double) MathHelper.sqrt(d0 * d0 + d2 * d2);
+        entityarrow.setThrowableHeading(d0,
+          d1 + d3 * GRAVITY_MULTIPLIER,
+          d2,
+          ARROW_VELOCITY,
+          (float) (DIFFICULTY_SUBTRACTER - this.world.getDifficulty().getDifficultyId() * DIFFICULTY_MULTIPLIER));
+        this.playSound(SoundEvents.ENTITY_SKELETON_SHOOT, 1.0F, 1.0F / (this.getRNG().nextFloat() * SOUND_MULTIPLIER + SOUND_ADDER));
+        this.world.spawnEntity(entityarrow);
+    }
+
+    protected EntityArrow getArrow(final float p_190726_1_)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)